drivers/power/pm/pm_procfs.c

```diff
@@ -181,6 +203,12 @@ static int pm_open(FAR struct file *filep, FAR const char *relpath,
       return -EACCES;
     }
 
+  relpath += strlen("pm/");
+  if (pm_path_validate(relpath) < 0)
```

Comment:
```suggestion
  i = pm_path_validate(relpath);
  if (i < 0)
```
and change line 221 to 231:
```
  pmfile->read = g_pm_files[i].read;
  pmfile->domain = atoi(relpath + strlen(g_pm_files[i].name));
```